client/state/data-layer/wpcom/me/settings/index.js

```diff
@@ -105,6 +110,16 @@ export const finishUserSettingsSave = ( { settingsOverride }, data ) => ( dispat
 	const userLibModule = require( 'calypso/lib/user' );
 	const userLib = userLibModule.default ? userLibModule.default : userLibModule; // TODO: delete line after removing add-module-exports.
 	userLib().fetch();
+
+	if ( onSuccess && typeof onSuccess === 'function' ) {
```

Comment:
Note: I added this as a possibility for `withFormBase` to call back in case user settings were saved successfully. It's used [here](https://github.com/Automattic/wp-calypso/blob/61565c5adf0e92095a711c65fcf0c153d27ba48e/client/me/form-base/with-form-base.jsx#L58). We usually don't do this and I think it isn't good practice. For now it's necessary to unlock the form so the user may navigate away after saving their changes. We should definitely find a better way to accomplish this behavior. Currently `withFormBase`, `protectForm` and the component itself are too tightly coupled and this may be a good candidate for refactoring after the reduxification.

Reply:
This. Let's add some `@TODO`s and make sure to address this problem as part of the reduxification work. Otherwise it might remain here forever 😉

Reply:
I guess this is related to your comment here https://github.com/Automattic/wp-calypso/pull/48730#discussion_r555621612, right? I made a note for myself but will add a comment to make sure it's won't be forgotten.

Reply:
Added via f82c4b6